homeassistant/components/huawei_lte/sensor.py

```diff
@@ -121,7 +122,22 @@ def setup_platform(
         data.subscribe(path)
         sensors.append(HuaweiLteSensor(data, path, SENSOR_META.get(path, {})))
 
-    add_entities(sensors, True)
+    # Pre-0.97 unique id migration. Old ones used the device serial number
+    # (see comments in HuaweiLteData._setup_lte for more info), as well as
+    # had a bug that joined the path str with periods, not the path components,
+    # resulting e.g. *_device_signal.sinr to end up as
+    # *_d.e.v.i.c.e._.s.i.g.n.a.l...s.i.n.r
+    entreg = await entity_registry.async_get_registry(hass)
+    for entid, ent in entreg.entities.items():
+        if ent.platform == "huawei_lte":
+            for sensor in sensors:
+                if ent.unique_id.endswith("_%s" % ".".join(sensor.path)):
```

Comment:
I guess we should start using f-strings now. Our old standard was new style string formatting, but f-strings are awesome.

At least we shouldn't use old style string formatting.

Reply:
Agreed on f-strings in general, not sure in this particular case but... meh, changed.

(Disagreed on old style formatting vs .format, in general but that's moot now here.)